packages/headless/doc-parser/src/controller-resolver.ts

```diff
@@ -60,8 +60,39 @@ function isFunction(item: ApiItem): item is ApiFunction {
   return item.kind === ApiItemKind.Function;
 }
 
-function extractTypesFromInitializer(entity: FuncEntity) {
-  return isObjectEntity(entity.returnType)
-    ? extractTypes(entity.returnType.members).types
-    : [];
+function extractTypesFrom(initializer: FuncEntity, utils: FuncEntity[]) {
+  const instanceTypes = extractTypesFromInitializerInstance(initializer);
+  const utilTypes = extractTypes(utils).types;
+  const types = instanceTypes.concat(utilTypes);
+
+  return removeDuplicates(types);
+}
+
+function extractTypesFromInitializerInstance(entity: FuncEntity) {
+  const {returnType} = entity;
+
+  if (typeof returnType === 'string') {
+    return [];
+  }
+
+  if (isObjectEntity(returnType)) {
+    return extractTypes(returnType.members).types;
+  }
+
+  return [];
+}
+
+function removeDuplicates(entities: ObjEntity[]) {
```

Comment:
Why can there be duplicates ?

Reply:
In the case of the Facet controller we have:

```
toggleSelect(selection: FacetValue): void;
isValueSelected(value: FacetValue): boolean;

get state() {
    return {
        values: FacetValue[]
    }
}
```

As the extractor explores/crawls the different branches, it will extract `FacetValue` three times.

Initially I considered not pushing into the array if the type is found. However, this would not catch duplicates extracted from another section.

```
// extract
Facet
    ... some FacetValue interfaces ...

Utils
    ... some more FacetValue interfaces

// result

Types
    FacetValue
```